dask_sql/cmd.py

```diff
@@ -103,11 +175,23 @@ def cmd_loop(
         if not text:
             continue
 
-        try:
-            df = context.sql(text, return_futures=False)
-            print(df)
-        except Exception:
-            traceback.print_exc()
+        meta_command_detected = meta_commands(text, context=context, client=client)
+
+        def sqlCaller(text):
+            """
+            This function is for showing progress bar
+            """
+            yield context.sql(text, return_futures=False)
+
+        if not meta_command_detected:
+            try:
+                with ProgressBar() as pb:
+                    for df in pb(sqlCaller(text), total=1, label="Executing"):
```

Comment:
Really good idea, to add a progress bar. But I think we can do even better here - having a progress bar that actually depends on the tasks on the cluster, similar to Dask's [progress bar](https://docs.dask.org/en/latest/diagnostics-distributed.html#progress-bar).
Two ideas here. In both cases you need to call `context.sql` with `return_futures=True`, as we need to interact with the Dask tasks. To turn the dataframe into wait-able tasks, call `.persist()` (because a dataframe is a lazy action - nothing to wait on).
Either we then actually use dask's progress bar. 
Or you use a combination of `dask.distributed.as_competed`  and `client.futures_of`, something like (untested):
```python
df = context.sql(text, return_futures=True)
# (include None handling here)
# Turn df into something awaitable:
df = df.persist()
# Now turn it into a list of futures
futures = client.futures_of(df)
# and finally show a progress bar:
for df in pb(as_completed(futures), total=len(futures), label="Executing"):
    continue
# In the end, show the actual result (which is now calculated, so that will be fast):
df = df.compute()
print(df.to_markdown(tablefmt="fancy_grid"))
```

Reply:
Thanks for the Suggestion 👍 really great idea, copy-pasted the above solution and it worked as expected. !!